src/serde_json.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{EvaluationError, EvaluationResult, StructValue, Value};
+
+impl TryFrom<serde_json::Value> for Value {
+    type Error = EvaluationError;
+
+    fn try_from(value: serde_json::Value) -> Result<Self, Self::Error> {
+        json_value_to_value(&value)
+    }
+}
+
+fn json_value_to_value(value: &serde_json::Value) -> EvaluationResult<Value> {
+    match value {
+        serde_json::Value::Bool(value) => Ok(Value::Bool(*value)),
+        serde_json::Value::Number(value) if value.is_i64() => {
+            Ok(Value::Int(value.as_i64().unwrap()))
```

Comment:
Not sure how common knowledge this is, but I had to look up if `is_i64` resulted in a guaranteed success when then doing `as_i64`. serde_json guarantees it.

Reply:
You are correct. Using `unwrap` especially in a library is sort of dangerous. I also checked the doc before writing this. :-)

The [serde_json::value::Number](https://docs.rs/serde_json/latest/serde_json/value/struct.Number.html) says:

> Returns true if the Number is an integer between i64::MIN and i64::MAX.
> 
> For any Number on which is_i64 returns true, as_i64 is guaranteed to return the integer value.
> 
So it should be safe.